pdns/dnsdistdist/docs/reference/dq.rst

```diff
@@ -303,6 +325,18 @@ This state can be modified from the various hooks.
     :param string raw: The raw string to be spoofed, e.g. `"\\192\\000\\002\\001"`.
     :param table raws: The raw strings to be spoofed, e.g. `{ "\\192\\000\\002\\001", "\\192\\000\\002\\002" }`.
 
+  .. method:: DNSQuestion:suspend(asyncID, queryID, timeoutMS) -> bool
+
+    .. versionadded:: 1.8.0
+
+    Suspend the processing for the current query, making it asynchronous. The query is then placed into memory, in a map called the Asynchronous Holder, until it is either resumed or the supplied timeout kicks in. The object is stored under a key composed of the tuple (`asyncID`, `queryID`) which is needed to retrieve it later, which can be done via :func:`getAsynchronousObject`.
```

Comment:
Definition of legal range of both IDs is missing.

Also I woulld have expect the query to be identified by an ID or IDs created by the Asynchronous Holder. If I understand correctly, the burden of creating a unique asyncID, queryID pair is now upon the caller of `suspend`?

Reply:
I'll clear up the range of both IDs, nice catch.
As for making the burden of creating unique pairs on the caller, I believe this is the most flexible way, as the Asynchronous holder has no clue about the logic behind the objects that are stored, while the caller does.